pkgs/stdenv/generic/setup.sh

```diff
@@ -144,6 +144,11 @@ exitHandler() {
         # - system time for all child processes
         echo "build time elapsed: " "${times[@]}"
     fi
+    if [ -n "${enablePhaseMetrics:-}" ] && [ -d "$out" ]; then
```

Comment:
Maybe error out if enablePhaseMetrics is set but out is not a directory.

Reply:
my concern there was if I error out, then the actual error won't get displayed for why `$out` wasn't created.